__tests__/api/getContests.js

```diff
@@ -2,17 +2,39 @@ import api from '../../src/api';
 import judges from '../../src/judges';
 jest.mock('../../src/judges');
 
-// I need more than one judge working to make better tests.
-test('concatenates contest lists correctly', () => {
+function blocked_object() {
   const {blocked, revoke} = Proxy.revocable({}, {});
   revoke();
+  return blocked;
+}
+
+// I need more than one judge working to make better tests.
+test('concatenates contest lists correctly', () => {
   // The blocked object has revoked access, that means none of its fields may be acessed.
   // This is to make sure getContests doesn't even look at any of its fields.
-  judges.codeforces.getContests.mockImplementation(() => Promise.resolve([blocked, blocked, blocked]));
-  return expect(api.getContests()).resolves.toEqual([blocked, blocked, blocked]);
+  const blocked = blocked_object();
```

Comment:
why is this necessary?
what is the problem of testing with actual `Contest` objects?
since this test is going to be changed soon, we will deal with this later

Reply:
I don't think there is a problem with using actual `Contest`. But this is more restrictive so I preferred it. We don't need to actually look to the contest object in that function. If in the future we need to, we can change the test.